File: atividade-4/029_intercal/029_intercal.hs
module Main where

-- Intercalar valores dos vetores

intercal :: [Int] -> [Int] -> [Int]
intercal xs [] = xs
intercal [] ys = ys
intercal (x:xs) (y:ys) = x:y:intercal xs ys

main = do
  a <- readLn :: IO [Int]
  b <- readLn :: IO [Int]
  print $ intercal a b

File: atividade-4/029_intercal/029_intercal.js
//Intercalar valores dos vetores

const list = [1, 3, 5, 7, 9];
const list2 = [2, 4, 6, 8, 10];

const intercal = ([x, ...xs], [y, ...ys]) => {
  if (!y) return xs;
  if (!x) return ys;

  return [x, y, ...intercal(xs, ys)];
};

console.log(intercal(list, list2));
